packages/lib/src/interactions/utils.ts

```diff
@@ -31,17 +32,26 @@ export function getRatioEndPoint(
   const originalRatio =
     Math.abs(endPoint.x - startPoint.x) / Math.abs(endPoint.y - startPoint.y);
 
-  if (originalRatio < ratio) {
-    return new Vector2(
-      startPoint.x + widthSign * height * ratio,
-      startPoint.y + heightSign * height
-    );
+  const shiftX = widthSign * (originalRatio < ratio ? height * ratio : width);
+  const shiftY = heightSign * (originalRatio < ratio ? height : width / ratio);
+
+  if (!clampCenter) {
+    return [
+      startPoint,
+      new Vector2(startPoint.x + shiftX, startPoint.y + shiftY),
+    ];
   }
 
-  return new Vector2(
-    startPoint.x + widthSign * width,
-    startPoint.y + (heightSign * width) / ratio
-  );
+  const centerPoint = endPoint
+    .clone()
+    .sub(startPoint)
+    .divideScalar(2)
+    .add(startPoint);
+
+  return [
+    new Vector2(centerPoint.x - shiftX / 2, centerPoint.y - shiftY / 2),
+    new Vector2(centerPoint.x + shiftX / 2, centerPoint.y + shiftY / 2),
+  ];
```

Comment:
.. but in the case where we want to keep the center at the same place, we distribute the shift between the `startPoint` and the `endPoint`